src/Chapter1.hs

```diff
@@ -554,7 +563,11 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Integral a => a -> a -> a -> a
+mid x y z | n == x = max y z
+          | n == y = max x z
+          | n == z = max x y
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = ...`, to tell the compiler, that your pattern matching is exhaustive 🙂
In this case, the last clause can be replaced with the `otherwise`, as if it came to this clause, there is no other way that `n` could be anything but `z` 💯